### DISCUSSION TITLE
isLoading is not set to true when using invalidateQueries

### DISCUSSION THREAD

**Discussion by CedricCouton** [TanStack/query#5146] [General]
Hi,

I want to detect the fact that the invalidateQueries has been called and behind the data are fetch to get fresh new data. In that case we don't have way to detect it. Indeed, the isLoading still false because this isLoading is based on the fact that there is no dataUpdatedAt. Also we could base on the isInvalidated of the query but it is not exposed in the QueryObserverBaseResult. Is there exist a way to detect it? Some idea : 
- Expose the isInvalidated in the QueryObserverBaseResult
- reinit the dataUpdatedAt when the invalidateQueries is called.

Thanks.

**Comment by TkDodo** [comment]
are you looking for `isFetching` maybe ?

**Reply by CedricCouton** [reply]
Yes but not that i want
